core/state/statedb.go

```diff
@@ -121,8 +124,8 @@ type StateDB struct {
 	SnapshotCommits      time.Duration
 
 	// record-replay: SubstatePreAlloc, SubstatePostAlloc, SubstateBlockHashes of StateDB
-	SubstatePreAlloc    substate.SubstateAlloc
-	SubstatePostAlloc   substate.SubstateAlloc
+	SubstatePreAlloc    substate.WorldState
+	SubstatePostAlloc   substate.WorldState
```

Comment:
Should we rename these fields accordingly? `PreWorldState` and `PostWorldState`?

Reply:
I agree. Please do so.

Reply:
This option is temporary, it is necessary for testing. At the moment we have not renamed the field data in the Substate module itself, because it breaks compatibility with already created databases (there is sensitivity in decoding).
This probably doesn't need to be merged into the main, but create a proper branch. I apologize for the mess.